nohost/src/NoHostServer.js

```diff
@@ -42,9 +43,22 @@ define(function (require, exports, module) {
      * instrumentation is enabled
      */
     NoHostServer.prototype.add = function (liveDocument) {
+        var id = liveDocument.doc.language.getId();
+
         if (liveDocument.setInstrumentationEnabled) {
-            // enable instrumentation
-            liveDocument.setInstrumentationEnabled(true);
+            //If new HTML document is empty, inject defaultHTML file
+            //so live preview doesn't break
+            if(liveDocument.doc.getText() === "" && id === "html") {
+                liveDocument.doc.setText(defaultHTML);
```

Comment:
When you create new document, it's empty and the live preview will try to parse the document but since there is not DOM it will complain and the live preview will break.